src/output-json.c

```diff
@@ -742,14 +743,58 @@ static int MacSetIterateToJSON(uint8_t *val, MacSetSide side, void *data)
     return 0;
 }
 
-static int CreateJSONEther(JsonBuilder *js, const Packet *p, const Flow *f)
+static int CreateJSONEther(
+        JsonBuilder *js, const Packet *p, const Flow *f, enum OutputJsonLogDirection dir)
 {
     if (p != NULL) {
         /* this is a packet context, so we need to add scalar fields */
         if (p->ethh != NULL) {
             jb_open_object(js, "ether");
-            uint8_t *src = p->ethh->eth_src;
-            uint8_t *dst = p->ethh->eth_dst;
+            uint8_t *src;
+            uint8_t *dst;
+            switch (dir) {
+                case LOG_DIR_FLOW:
+                    if
+                        PKT_IS_TOCLIENT(p)
+                        {
+                            src = p->ethh->eth_dst;
+                            dst = p->ethh->eth_src;
+                        }
+                    else {
+                        src = p->ethh->eth_src;
+                        dst = p->ethh->eth_dst;
+                    }
+                    break;
+                case LOG_DIR_FLOW_TOCLIENT:
+                    if
+                        PKT_IS_TOSERVER(p)
+                        {
+                            src = p->ethh->eth_dst;
+                            dst = p->ethh->eth_src;
+                        }
+                    else {
+                        src = p->ethh->eth_src;
+                        dst = p->ethh->eth_dst;
+                    }
+                    break;
+                case LOG_DIR_FLOW_TOSERVER:
+                    if
+                        PKT_IS_TOCLIENT(p)
+                        {
+                            src = p->ethh->eth_dst;
+                            dst = p->ethh->eth_src;
+                        }
+                    else {
+                        src = p->ethh->eth_src;
+                        dst = p->ethh->eth_dst;
+                    }
```

Comment:
Weird formatting, but it passes the check.  When I run the formatting scripts it reformats it back to something more normal.

Reply:
I don't know what happened here. I used regular formatting and it was changed to that by clang.